tvaultuiv2/src/services/helper-function.js

```diff
@@ -40,19 +40,29 @@ export function findElementById(arr, id, nestingKey) {
     'Not found'
   );
 }
+export const findItemAndRemove = (arr, key, id) => {
+  if (arr?.length === 0) return;
+  const tempArr = [...arr];
+  const indexofItem = tempArr[0][key].findIndex((item) => item.id === id);
+  tempArr[0][key].splice(indexofItem, 1);
+  return tempArr;
+};
 
 export const findElementAndUpdate = (arr, parentId, item) => {
-  if (arr.length === 0) return;
+  if (arr?.length === 0) return;
```

Comment:
```suggestion
  if (arr?.length === 0) return;
```

Array.isArray(arr) && arr.length === 0